### PR TITLE
util: remove path deps

### PR DESCRIPTION
**Pull request by taiki-e**

### CODE REVIEW

File: tokio-stream/src/wrappers/lines.rs [hunk 1]
```diff
@@ -30,6 +30,7 @@ impl<R> LinesStream<R> {
     }
 
     /// Obtain a pinned reference to the inner `Lines<R>`.
+    #[allow(clippy::wrong_self_convention)] // TODO: rename to get_pin_mut
     pub fn as_pin_mut(self: Pin<&mut Self>) -> Pin<&mut Lines<R>> {
```
Comment (Darksonn) [inline tokio-stream/src/wrappers/lines.rs:34]: I'm not convinced these should be renamed. The clippy lint just doesn't handle this case, and allows `&mut self`. All the other methods on this type are, by virtue of `AsRef`/`AsMut`, also called `as_*`.

Comment (Darksonn) [reply tokio-stream/src/wrappers/lines.rs:34]: In fact, there's an open issue for it. https://github.com/rust-lang/rust-clippy/issues/4546

Comment (taiki-e) [reply tokio-stream/src/wrappers/lines.rs:34]: Good catch, that's certainly not a right clippy warning. I will replace todo with a link to the clippy issue.

(IIRC all other tokio crates name similar methods `get_pin_mut`, so I personally tend to prefer to rename it for consistency whether clippy is right or not. But that's not the topic I want to deal with in this PR...)

**Comment by taiki-e** [comment]
Ok, I think I fixed all "remove path deps"-related errors that @Darksonn encountered in https://github.com/tokio-rs/tokio/pull/3402.

**Review by Darksonn — APPROVED** [review]
If clippy has no further complaints, then this looks good to me.

**Comment by taiki-e** [comment]
Running nightly clippy gives me some warnings that break code if accepting the suggestion, but in MSRV it's fine.

**Pull request merged**